Tasks/Npm/npmtask.ts

```diff
@@ -87,6 +97,16 @@ async function runNpmAuthHelperAsync(npmAuthRunner: trm.ToolRunner) : Promise<nu
     }
 }
 
+function getUserNpmrcPath() {
+    var userNpmRc = process.env['npm_config_userconfig'];
+    if(!userNpmRc){
+        // default npm rc is located at user's home folder.
+        userNpmRc = path.join(process.env['HOMEDRIVE'], process.env['HOMEPATH'], '.npmrc');
```

Comment:
there was a cost to the auth helper conversion to .net core we couldn't absorb, plus the desire to keep it all working on dev boxes and with existing identity stack (so not just build agents).  but, agree.  Here, are you saying Homedrive and homepath are bad assumptions here and there is a better common way across arch to get the user profile from the task lib?  I'd guess no since I'd guess they don't want anyone touching the user profile for an agent :)